src/compiler/binder.ts

```diff
@@ -429,6 +431,9 @@ module ts {
                 case SyntaxKind.TypeParameter:
                     bindDeclaration(<Declaration>node, SymbolFlags.TypeParameter, SymbolFlags.TypeParameterExcludes, /*isBlockScopeContainer*/ false);
                     break;
+                case SyntaxKind.TypeGuardType:
+                    bindDeclaration(<Declaration>node, (SymbolFlags.TypeGuardType | SymbolFlags.Value), SymbolFlags.TypeGuardTypeExcludes, /*isBlockScopeContainer*/ false);

```

Comment:
why is that even bound? it does not create a symbol.. it is not creating a new type, it is just `boolean` with additional properties.. correct?